### PR TITLE
Migrated to ArcGIS Runtime 10.2.6-2 and Android Studio

### LINKED ISSUE #104: Update to ArcGIS Runtime 10.2.6 and Android Studio
The biggest impact here is Android Studio, which is the IDE supported by ArcGIS Runtime 10.2.5 and higher for Android. Migrating from Eclipse to Android Studio moves lots of files around and creates new files. In my opinion, this should be the first task to complete after Squad Leader 3.0.0 is finalized.

### PR DESCRIPTION
**Pull request by garys-esri**
Fixes #104

This is a big pull request because we had to move a lot of files around to migrate from Eclipse to Android Studio. We also removed Eclipse-specific files and added Android Studio-specific files.

@csmoore, if you have time in the next week, please take a look and merge.

### CODE REVIEW

**Comment by garys-esri** [comment]
Actually @csmoore, hold off please. I have a couple of things I need to do first.

**Comment by csmoore** [comment]
Sure - if it can wait until Monday, I will have some time to actually try to set up the dev environment to test out the build

**Comment by garys-esri** [comment]
Part of what I need to do is document how to build it, so hopefully it will be relatively easy for you.

**Comment by garys-esri** [comment]
@csmoore, this is ready to check when you are ready. Thanks in advance.

**Comment by csmoore** [comment]
Sure, I'll have a look - under the guise if 'if I can do it anyone can' - just a heads up I may not get to until next Monday (I am out one of the days)

**Comment by csmoore** [comment]
Not part of these updates but the prebuilt Download link at the [beginning of the readme](https://github.com/garys-esri/squad-leader-android#squad-leader-android) http://www.arcgis.com/home/group.html?owner=Arcgisonline_defense&title=ArcGIS%20for%20Defense%20and%20Intelligence looks like it might be the old one vs. http://www.esri.com/apps/products/download/#ArcGIS_for_the_Military

**Comment by garys-esri** [comment]
Good catch. I updated it with a better link and text in 7e8bed7583e8214f204dc9c9d231186a25f242e5 .

**Comment by csmoore** [comment]
I tried the gradle build first & got this error
```
> squad-leader-android\source\SquadLeader>gradlew.bat assemble

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':aFileChooser'.
> failed to find target android-19 : C:\Users\chri0020\AppData\Local\Android\sdk

BUILD FAILED
```

I have -23 installed - do I need to install older SDK versions and/or update some configuration file somewhere?:

image

File: README.md [hunk 1]
```diff
@@ -71,83 +84,42 @@ To build the app from source:
...
+1. In the command prompt, set the ANDROID_HOME environment variable to the location of your Android SDK installation. This is likely to be `C:\Users\<username>\AppData\Local\Android\sdk`. (You can set this variable in your system before starting the command prompt if desired.)
+1. On Windows, run `gradlew.bat assemble`. On Linux, run `./gradlew assemble`.
+1. When the build successfully completes, `.apk` files are written to `source/SquadLeader/app/build/outputs/apk`.
+
+## Running Unit Tests
+
+### Running Unit Tests using Android Studio
+
+1. Follow the [Building from Source](#building-from-source) and [Building with Android Studio](#building-with-android-studio) steps above, except that there is no need to run the app or generate an APK file.
+1. In Android Studio, in the Select Debug/Run Configuration dropdown, change from **app** to **All Tests**.
```
Comment (csmoore) [inline README.md]: I did not have this option (just "app") - is there something I needed to do first (besides build app)?

File: README.md [hunk 2]
```diff
@@ -21,21 +21,34 @@ The Squad Leader template demonstrates best practices for building handheld mili
...
 To build the app from source:
 
 - Android SDK
-  - Android API 14 or higher
-- [ArcGIS Runtime SDK 10.2.4 for Android](https://developers.arcgis.com/en/downloads/)
-  - Even if you have installed the Eclipse plugin from the online update site, you'll need to download the SDK to get the native binary files (see [build steps](#building-with-eclipse)).
-- Eclipse 3.6.2 or higher (Eclipse for Android recommended)
-- Android Support Library
+  - Android API 15 or higher
+  - Android Support Repository and Android Support Library must be installed, using the Android SDK Manager.
```
Comment (csmoore) [inline README.md]: I initially didn't know what this meant or how to check, but someone more familiar would Android likely would (looks like the Support Library wasn't installed by default, not sure if that explains any of my issues):

image

File: README.md [hunk 3]
```diff
@@ -21,21 +21,34 @@ The Squad Leader template demonstrates best practices for building handheld mili
...
 
 To build the app from source:
 
 - Android SDK
-  - Android API 14 or higher
-- [ArcGIS Runtime SDK 10.2.4 for Android](https://developers.arcgis.com/en/downloads/)
-  - Even if you have installed the Eclipse plugin from the online update site, you'll need to download the SDK to get the native binary files (see [build steps](#building-with-eclipse)).
-- Eclipse 3.6.2 or higher (Eclipse for Android recommended)
-- Android Support Library
+  - Android API 15 or higher
```
Comment (csmoore) [inline README.md]: There are some other/higher APIs mentioned in the build/config files (and there is a viewshed comment that mentions another) so just wanted to confirm this Android API value

File: README.md [hunk 4]
```diff
@@ -161,7 +133,7 @@ To build the app from source:
 Squad Leader leverages the following as submodules of the squad-leader-android repository. 
 
 * [military-apps-library-java](https://github.com/Esri/military-apps-library-java) 
-* [aFileChooser](https://github.com/iPaulPro/aFileChooser)
+* [aFileChooser](https://github.com/garys-esri/aFileChooser)
  
 If you should want to update to the latest commit instead of the commit used by the squad-leader-android commit you're using, you can open a GitHub shell in squad-leader-android and run the following:
```
Comment (csmoore) [inline README.md]: This version might work better for multiple submodules: `> git submodule foreach git pull origin master`

File: README.md [hunk 5]
```diff
@@ -71,83 +84,42 @@ To build the app from source:
...
+1. In Android Studio, click the **Sync Project with Gradle Files** button, or choose **Tools > Android > Sync Project with Gradle Files**. Wait a few seconds for the Gradle sync to complete.
+1. In Android Studio, click the **Make Project** button, or type Ctrl-F9. Wait a few seconds for the build to complete.
+1. To run the app, connect an Android device by USB, then click the **Run** button, or type Shift-F10.
+1. To generate an Android installer (APK) file, go to **Build > Generate Signed APK**.
+
+### Building with Gradle
+
+1. Open a command prompt and navigate to your clone. Inside your clone, navigate to `source/SquadLeader`.
+1. In the command prompt, set the ANDROID_HOME environment variable to the location of your Android SDK installation. This is likely to be `C:\Users\<username>\AppData\Local\Android\sdk`. (You can set this variable in your system before starting the command prompt if desired.)
+1. On Windows, run `gradlew.bat assemble`. On Linux, run `./gradlew assemble`.
```
Comment (csmoore) [inline README.md]: Note: this works now after adding the additional dependencies through the SDK manager (android-19 and build tools)

**Comment by csmoore** [comment]
I wasn't able to get the default Nexus 5 emulator to run - but I can confirm the build works fine. The only issues I had/noticed that you might want to address:

- [ ] those 2 dependencies (android-19 & Build Tools 21)
- [ ] add any additional steps needed to run the unit test 
- [ ] perhaps get someone to sanity check the apk since I wasn't able to run/test

It is ready to merge after you decide if/what to do with those issues

**Comment by garys-esri** [comment]
We actually only need Android API level 15 for this app, so I changed the requirement to that in 553320830359b7fcc6ba08abde31b884ce0c577f . I will document that and the build tools requirement, with a note that you can modify build.gradle if you want to use newer versions. Sorry that was unclear.

**Comment by garys-esri** [comment]
I clarified the dependencies in e74733f0c1ca67b38f13141de5e9d69f27aff652.

**Comment by garys-esri** [comment]
On the missing All Tests, that's my mistake. It's stored in .idea/workspace.xml, which is in .gitignore and which IntelliJ [recommends](https://intellij-support.jetbrains.com/hc/en-us/articles/206827587-How-to-manage-projects-under-Version-Control-Systems) be ignored by version control systems. So I will instead add documentation on how to run unit tests.

**Comment by garys-esri** [comment]
I added additional steps for unit tests in e854068508cd6a83d9cfc589f0ce534ef33aa3d4.

@csmoore, unless you object, I would suggest that we save the final user testing with an .apk until we're closer to an actual release. If that's agreeable, I think we're ready to merge unless I missed anything. Thanks again for your great review.

**Comment by csmoore** [comment]
Sure, sounds good - I just wanted to let you know I hadn't actually run - merging now

**Pull request merged**